### PR TITLE
TileHandling

### PR DESCRIPTION
**Pull request by gitZ0N3**

### CODE REVIEW

File: src/main/java/carcassone/client/desktop/Controller.java [hunk 1]
```diff
@@ -41,10 +73,129 @@ public void initialize(){
...
+
+            dragPane.toFront();
+            dragging = true;
+        }
+
+        if (!isMeeple(image)) {
+            dragTileImage.setX(event.getX() - 50);
+            dragTileImage.setY(event.getY() - 50);
+        } else {
+            dragMeepleImage.setX(event.getX() - 16);
```
Comment (verebd) [inline src/main/java/carcassone/client/desktop/Controller.java]: same question for -16

File: src/main/java/carcassone/client/desktop/Controller.java [hunk 2]
```diff
@@ -41,10 +73,129 @@ public void initialize(){
...
+            dragTileImage.setImage(image);
+            dragMeepleImage.setImage(image);
+            tileFrom.setImage(null);
+
+            dragPane.toFront();
+            dragging = true;
+        }
+
+        if (!isMeeple(image)) {
+            dragTileImage.setX(event.getX() - 50);
```
Comment (gitZ0N3) [inline src/main/java/carcassone/client/desktop/Controller.java]: It is half of the node width/height. This way it looks like the tile/meeple is dreagged in the middle of the image. I've changed it to be calculated automatically:

        if (!isMeeple(image)) {
            dragTileImage.setX(event.getX() - dragTileImage.getFitWidth()/2);
            dragTileImage.setY(event.getY() - dragTileImage.getFitHeight()/2);
        } else {
            dragMeepleImage.setX(event.getX() - dragMeepleImage.getFitWidth()/2);
            dragMeepleImage.setY(event.getY()- dragMeepleImage.getFitHeight()/2);
        }

File: src/main/java/carcassone/client/desktop/Controller.java [hunk 3]
```diff
@@ -41,10 +73,129 @@ public void initialize(){
...
+    }
+
+    public void exitMouseDrag() {
+        if (dragging) {
+            if(!isMeeple(image))
+                setNodeOrder("tileGrid");
+            if (isMeeple(image))
+                setNodeOrder("meepleGrid");
+
+            dragTileImage.setX(-100);
```
Comment (verebd) [inline src/main/java/carcassone/client/desktop/Controller.java]: What does it mean here that the coordinates of the meeple and tiles  are set to (-100,-100)?

Comment (gitZ0N3) [inline src/main/java/carcassone/client/desktop/Controller.java]: Two tiles are created to simulate dragging. After exiting dragging these tiles will be moved outside of the gameboard to (-100, -100). This way they could not be clicked accidentally. 

An other solution would be to move the node back but I found the first one more simple.

**Pull request merged**